File: POH7/Megane/Main.cpp
// Try POH
// author: Leonardone @ NEETSDKASU
#include <iostream>

using namespace std;

int main() {
    int q[100][100];
    int p[10][10];
    int n, m;
    
    cin >> n;
    for (int i = 0; i < n; i++)
        for (int j = 0; j < n; j++)
            cin >> q[i][j];
    cin >> m;
    for (int i = 0; i < m; i++)
        for (int j = 0; j < m; j++)
            cin >> p[i][j];
    
    for (int i = 0; i <= n - m; i++)
        for (int j = 0; j <= n - m; j++) {
            int c = 0;
            for (int y = 0; y < m; y++) 
                for (int x = 0; x < m; x++)
                    c |= q[i + y][j + x] ^ p[y][x];
            if (!c)
                cout << i << " " << j << endl;
        }
    
    return 0;    
}

File: POH7/Megane/Main.scala
/* Try POH
 * author: Leonardone @ NEETSDKASU
 */
object Main extends App {
    
    def gs() = scala.io.StdIn.readLine
    def gi() = gs.toInt
    def gss() = gs.split(' ')
    def gis() = gss.map( _.toInt )
    def ngt[T](n :Int, f: ()=>T) = (1 to n).map( (_) => f() )
    def ngs(n :Int) = ngt(n, gs)
    def ngi(n :Int) = ngt(n, gi)
    def ngss(n :Int) = ngt(n, gss)
    def ngis(n :Int) = ngt(n, gis)
    
    val n = gi
    val q = ngis(n)
    val m = gi
    val p = ngis(m)
   
    for (i <- 0 to n - m)
        for (j <- 0 to n - m)
            if ( (0 until m).flatMap( (y) => (0 until m).map( (x) => q(i+y)(j+x) == p(y)(x) ) ).reduce( _ & _ ) )
                Console.println(List(i, j).mkString(" "))

}
